Docs: Update transform docs

nmarrs opened the pull request:
Update transform docs for enum convert field support. Also add docs for `Other` convert field type option (see [this PR for context on this option](https://github.com/grafana/grafana/pull/42624))
<img width="483" alt="Screenshot 2023-11-16 at 12 31 36 PM" src="https://github.com/grafana/grafana/assets/22381771/efbfb02d-ebd2-4533-a31a-53013f6f0f72">

Related to https://github.com/grafana/grafana/pull/76410

nmarrs committed:
update transform docs

imatwawana reviewed (commented):
This LGTM, however, I'm pretty sure you should be updating the following file public/app/features/transformers/docs/content.ts (it's noted in the front matter of this file). @baldm0mma, yes?

baldm0mma commented:
> This LGTM, however, I'm pretty sure you should be updating the following file public/app/features/transformers/docs/content.ts (it's noted in the front matter of this file). @baldm0mma, yes?

That's correct, thanks @imatwawana! @nmarrs the markdown is now being built from the content in `public/app/features/transformers/docs/content.ts` - we did this so that the documentation can now be available in the app itself for referencing. Please update the documentation in `content.ts`, then run the script mentioned at the top of that file to rebuild the markdown in the `index.md`. Then we'll merge both files.

baldm0mma commented on docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:
```diff
@@ -272,6 +272,9 @@ Use this transformation to change the field type of the specified field.
   - **Time -** attempts to parse the values as time
     - Will show an option to specify a DateFormat as input by a string like yyyy-mm-dd or DD MM YYYY hh:mm:ss
   - **Boolean -** will make the values booleans
+  - **Enum -** will make the values enums
+    - Will show a table to manage the enums
+  - **Other -** attempts to parse the values as JSON
```
```suggestion
  - **Enum** - will make the values enums
    - Will show a table to manage the enums
  - **Other** - attempts to parse the values as JSON
```
Just made a quick change to the new syntax, and again, this should be updated in the `content.ts` file, the script mentioned at the top of the file run, then the markdown should reflect the change.

nmarrs committed:
update PR based on feedback; update formatting of list items for the entire file

nmarrs commented:
Ah was not aware that some docs are controlled by TS files 😬 thanks for the heads up! I've updated the PR accordingly and also fixed the list item formatting for the entire file :)

imatwawana reviewed (approved):
LGTM!
